### PR TITLE
OCM-5576 | fix:  Edit autoscaling max replicas of nodepool force to s…

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/OCM-5576

### CODE REVIEW

File: cmd/edit/machinepool/nodepool.go [hunk 1]
```diff
@@ -307,12 +307,25 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
 	asBuilder := cmv1.NewNodePoolAutoscaling()
 	changed := false
 
-	if nodePool.Autoscaling().MinReplica() != minReplicas && minReplicas >= 1 {
-		asBuilder = asBuilder.MinReplica(minReplicas)
+	existingMinReplica := nodePool.Autoscaling().MinReplica()
+	existingMaxReplica := nodePool.Autoscaling().MaxReplica()
+
+	if existingMinReplica != minReplicas {
+		if minReplicas >= 1 {
```
Comment (marcolan018) [inline cmd/edit/machinepool/nodepool.go:314]: should `changed = true` only when `minReplicas >= 1`?

File: cmd/edit/machinepool/nodepool.go [hunk 2]
```diff
@@ -307,12 +307,25 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
...
+		}
+		if minReplicas == 0 {
+			asBuilder = asBuilder.MinReplica(existingMinReplica)
+		}
 		changed = true
 	}
-	if nodePool.Autoscaling().MaxReplica() != maxReplicas && maxReplicas >= 1 {
-		asBuilder = asBuilder.MaxReplica(maxReplicas)
+	if existingMaxReplica != maxReplicas {
+		if maxReplicas >= 1 {
```
Comment (marcolan018) [inline cmd/edit/machinepool/nodepool.go:323]: same as above, `changed = true` should not be here if maxReplicas not set?

Comment (chenz4027) [reply cmd/edit/machinepool/nodepool.go:323]: Thanks @marcolan018 , updated with the change

File: cmd/edit/machinepool/nodepool.go [hunk 3]
```diff
@@ -307,13 +307,26 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
...
-	if nodePool.Autoscaling().MaxReplica() != maxReplicas && maxReplicas >= 1 {
-		asBuilder = asBuilder.MaxReplica(maxReplicas)
-		changed = true
+	if existingMaxReplica != maxReplicas {
+		if maxReplicas >= 1 {
+			asBuilder = asBuilder.MaxReplica(maxReplicas)
+			changed = true
+		}
+		if maxReplicas == 0 {
+			asBuilder = asBuilder.MaxReplica(existingMaxReplica)
```
Comment (gdbranco) [inline cmd/edit/machinepool/nodepool.go:328]: Would it be ok to set it to the existing values initially instead to save on some condition branching?

Comment (chenz4027) [reply cmd/edit/machinepool/nodepool.go:328]: Haha I actually did that first and refactored it to what it looks like now - will make the change

File: cmd/edit/machinepool/nodepool.go [hunk 4]
```diff
@@ -307,13 +307,27 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
...
+		max = existingMaxReplica
+	} else {
+		max = maxReplicas
+	}
+
+	if existingMinReplica != minReplicas || existingMaxReplica != maxReplicas {
+		if min >= 1 && max >= 1 {
+			asBuilder = asBuilder.MinReplica(min).MaxReplica(max)
+			changed = true
+		}
```
Comment (ziccardi) [inline cmd/edit/machinepool/nodepool.go:330]:
```suggestion
		if min >= 1 && max >= 1 {
			return cmv1.NewNodePoolAutoscaling().MinReplica(min).MaxReplica(max)
		}
```

the `asBuilder` is returned only if the code enters this path. We can return it directly here and get rid of the `asBuilder` and `changed` variables declared at the start of the method.

Comment (chenz4027) [reply cmd/edit/machinepool/nodepool.go:330]: Thanks @ziccardi , will make the change

**Comment by chenz4027** [comment]
@ziccardi I updated the refactor.

@gdbranco Provisioned a hcp and tested switching autoscaling on and off. 

```
./rosa edit machinepool workers -c magchen-hcp --enable-autoscaling --min-replicas 2 --max-replicas 6
I: Updated machine pool 'workers' on hosted cluster 'magchen-hcp'
./rosa edit machinepool workers -c magchen-hcp -i                        
? Enable autoscaling: No
? Replicas: 2
? Labels (optional): 
? Taints (optional): 
? Autorepair: Yes
? Node drain grace period (optional): 
I: Updated machine pool 'workers' on hosted cluster 'magchen-hcp'
```

Looks fine 
```
./rosa describe cluster -c magchen-hcp

Name:                       magchen-hcp
Domain Prefix:              magchen-hcp
Display Name:               magchen-hcp
...
Nodes:
 - Compute (desired):       2
 - Compute (current):       2

```

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**